go.mod

```diff
@@ -89,6 +88,7 @@ require (
 	github.com/prometheus/client_model v0.3.0 // indirect
 	github.com/prometheus/common v0.37.0 // indirect
 	github.com/prometheus/procfs v0.8.0 // indirect
+	github.com/sirupsen/logrus v1.9.0 // indirect
```

Comment:
I think we discussed logging when this example-driver was initially created and concluded to defer further work on that.

My own preference would be to do logging as in other Kubernetes components, using k8s.io/component-base/logs, including JSON support.

This doesn't need to be done in this PR, but perhaps you want to pick that up as a next step?

Looking at this diff, there are some other dependencies in go.mod that are questionable: do we really need github.com/spf13/viper?

/cc @byako @klueska

Reply:
> This doesn't need to be done in this PR, but perhaps you want to pick that up as a next step?

Yeah, I'll work on that in another PR.

Reply:
@pohly I think it's a matter of preference. It does not seem to be vital, but there must have been some convenience reason that it was added.

Reply:
The code needs some logging solution. logrus used to be popular, but it's not a good choice anymore.

@klueska needs to confirm, but I suspect using it wasn't really a conscious decision and "just happened".

Let's discuss in https://github.com/kubernetes-sigs/dra-example-driver/issues/10

Reply:
I don't have a preference for logging.

The viper dependency comes from the desire to use environment variables to pass options rather than command line flags. From what I could tell viper was the standard way to integrate with cobra for this functionality.

At NVIDIA we typically use [urfave/cli](https://github.com/urfave/cli) which has this functionality built in, but the go-client libraries use cobra, and this is a kubernetes-sigs project, so I figured it was better to stick with that.